src/frequenz/sdk/timeseries/_moving_window.py

```diff
@@ -67,17 +70,20 @@ class MovingWindow:
     time_end = time_start + timedelta(minutes=5)
 
     # ... wait for 5 minutes until the buffer is filled
-    await asyncio.sleep(5)
-
-    # return an numpy array from the window
-    a = window[time_start:time_end]
-    # and use it to for example calculate the mean
-    mean = a.mean()
-    '''
+    await asyncio.sleep(0.1)
+
+    try:
+        # return an numpy array from the window
+        a = window[time_start:time_end]
+        # and use it to for example calculate the mean
+        mean = a.mean()
+    except IndexError as e:
+        pass
+    ```
 
     **Example2** (create a polars data frame from a `MovingWindow`):
 
-    ```
+    ``` disabled-no-polars-installed
```

Comment:
Is this something you are interpreting yourself? If so, a few comments:

1. It would be interesting to add a *directive* to `pip install` extra dependencies for tests.
2. We should probably still make the code block a normal python code block, so it is rendered properly when the documentation is rendered
3. We should probably implement the *directives* as markdown comments do they don't show on the rendered docs, or even better, look for some markdown format that would render nicely, as it is probably important info for the people reading the docs that they need to install some extra dependency to try this example. Maybe it can be a special comment in the python example, like:

   ```python
   # example-pip-dependencies: polars
   import polars as pl
   ```

   Then the runner will do a `pip install polars` before running the test.

Reply:
Nah, it's actually not interpreted at all, it just doesn't match the regex and thus is ignored. I thought about installing deps like that but didn't wanna expand the required installed dependencies for tests at this point.

I did think about directives as well, yes.